test/tests/ics/dependency/monomial.i

```diff
@@ -86,32 +86,36 @@
 
 [BCs]
   [./left_u]
-    type = DirichletBC
+    type = PenaltyDirichletBC
     variable = u
     boundary = left
     value = 0
+    penalty = 1000
   [../]
 
   [./right_u]
-    type = DirichletBC
+    type = PenaltyDirichletBC
     variable = u
     boundary = right
     value = 1
+    penalty = 1000
   [../]
 
 
   [./left_v]
-    type = DirichletBC
+    type = PenaltyDirichletBC
     variable = v
     boundary = left
     value = 2
+    penalty = 1000
   [../]
 
   [./right_v]
-    type = DirichletBC
+    type = PenaltyDirichletBC
     variable = v
     boundary = right
     value = 1
+    penalty = 1000
```

Comment:
These changes look unrelated to your PR. Why was this changed?

Reply:
With my changes this test errors out. I suppose there are a couple of ways to think about DirichletBC. If we think about it as assigning values for DOFs then I suppose my changes should be compatible with this test as it stands in HEAD. However, if we think about them as assigning values on domain boundaries then I don't think using a DirichletBC makes sense in the context of discontinuous monomials.